assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationIntroductoryOverlayNotification.test.js

```diff
@@ -50,22 +55,37 @@ describe( 'AudienceSegmentationIntroductoryOverlayNotification', () => {
 
 	beforeAll( () => {
 		registry = createTestRegistry();
+		provideUserInfo( registry );
+		provideModules( registry, [
+			{
+				slug: 'analytics-4',
+				active: true,
+				setupComplete: true,
+			},
+		] );
 	} );
 
 	it( 'should render an introductory overlay notification', async () => {
 		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
 
-		const { getByText, waitForRegistry } = render(
+		const userID = registry.select( CORE_USER ).getID();
+
+		registry
+			.dispatch( MODULES_ANALYTICS_4 )
+			.setAudienceSegmentationSetupCompletedBy( userID + 1 );
+
+		const { queryByText, waitForRegistry } = render(
 			<AudienceSegmentationIntroductoryOverlayNotification />,
 			{
 				registry,
+				context: VIEW_CONTEXT_MAIN_DASHBOARD,
 			}
 		);
 
 		await waitForRegistry();
 
 		expect(
-			getByText(
+			queryByText(
```

Comment:
A trivial detail - when we _do_ expect an element to exist, we do use `getByText()` rather than `queryByText()`, as `getByText()` is slightly safer as it throws an error if an element with matching text isn't found.

```suggestion
			getByText(
```

Indeed, strictly speaking we don't even need to test for `toBeInTheDocument()` when using `getByText()` as the test will fail if the text isn't found. However we do have an informal convention of still checking for `toBeInTheDocument()`. I suppose it makes the test a bit more explicit.